marlowe-playground-client/grammar.ne

```diff
@@ -0,0 +1,180 @@
+@{%
+const moo = require("moo");
+
+const lexer = moo.compile({
+        WS: /[ \t]+/,
+        number: /0|-?[1-9][0-9]*/,
+        string: {match: /"(?:\\["\\]|[^\n"\\])*"/, value: x => x.slice(1, -1)},
+        comma: ',',
+        lparen: '(',
+        rparen: ')',
+        lsquare: '[',
+        rsquare: ']',
+        hole: /\?[a-zA-Z0-9_-]+/,
+        CONSTRUCTORS: {
+            match: /[A-Z][A-Za-z]+/, type: moo.keywords({
+                CONTRACT: ['Close', 'Pay', 'If', 'When', 'Let'],
+                OBSERVATION: [
+                    'AndObs',
+                    'OrObs',
+                    'NotObs',
+                    'ChoseSomething',
+                    'ValueGE',
+                    'ValueGT',
+                    'ValueLT',
+                    'ValueLE',
+                    'ValueEQ',
+                    'TrueObs',
+                    'FalseObs',
+                ],
+                VALUE: [
+                    'AvailableMoney',
+                    'Constant',
+                    'NegValue',
+                    'AddValue',
+                    'SubValue',
+                    'ChoiceValue',
+                    'SlotIntervalStart',
+                    'SlotIntervalEnd',
+                    'UseValue',
+                ],
+                ACCOUNT_ID: ['AccountId'],
+                TOKEN: ['Token'],
+                PAYEE: ['Account', 'Party'],
+                PARTY: ['PK', 'Role'],
+                BOUND: ['Bound'],
+                VALUE_ID: ['ValueId'],
+                CASE: ['Case'],
+                ACTION: ['Deposit', 'Choice', 'Notify'],
+                CHOICE_ID: ['ChoiceId'],
+            })
+        },
+        NL: { match: /\n/, lineBreaks: true },
+});
+
+%}
+
+# Pass your lexer object using the @lexer option:
+@lexer lexer
+
+main -> topContract ws:* {%([contract,]) => contract%}
```

Comment:
Probably would be good to add `ws` before `topContract` too

Reply:
I've added this but just discovered that coupon bond guaranteed seems to be broken in alpha, I think due to linter. Will check after this PR.

Reply:
yes, looks like we need to separate editor errors and editor warnings. I will do this in another PR.